src/API/WesternStatesWater.WestDaat.Engines/ValidationEngine.cs

```diff
@@ -0,0 +1,116 @@
+using WesternStatesWater.Shared.DataContracts;
+using WesternStatesWater.Shared.Errors;
+using WesternStatesWater.WestDaat.Common.Context;
+using WesternStatesWater.WestDaat.Common.Extensions;
+using WesternStatesWater.WestDaat.Contracts.Client.Requests.Conservation;
+using WesternStatesWater.WestDaat.Utilities;
+
+namespace WesternStatesWater.WestDaat.Engines;
+
+internal class ValidationEngine : IValidationEngine
+{
+    private readonly IContextUtility _contextUtility;
+
+    private readonly ISecurityUtility _securityUtility;
+
+    public ValidationEngine(
+        IContextUtility contextUtility,
+        ISecurityUtility securityUtility
+    )
+    {
+        _contextUtility = contextUtility;
+        _securityUtility = securityUtility;
+    }
+
+    public async Task<ErrorBase> Validate<TRequest>(TRequest request) where TRequest : RequestBase
+    {
+        var context = _contextUtility.GetContext();
+        var permissions = await _securityUtility.GetPermissions(context);
+
+        return request switch
+        {
+            ApplicationStoreRequestBase req => ValidateUserLoadRequest(req, context, permissions),
+
+            _ => throw new NotImplementedException(
+                $"Validation for request type '{request.GetType().Name}' is not implemented."
+            )
+        };
+    }
+
+    private ErrorBase ValidateUserLoadRequest(ApplicationStoreRequestBase request, ContextBase context,
```

Comment:
```suggestion
    private ErrorBase ValidateApplicationStoreRequest(ApplicationStoreRequestBase request, ContextBase context,
```

I'm also assuming this method will contain an additional switch. `RequestBase -> ApplicationStoreRequestBase -> FutureApplicationStoreWeDontHaveYetRequest`

Reply:
We could do multiple switches or we could do one big switch. I just didn't want to add any more DTOs so I used the base for the example.